File: backend/src/Modules/Travel/Dtos/ItineraryDto.cs
namespace Trippie.Modules.Travel.Dtos;

public sealed class CreateItineraryDto
{
    public required int TripId { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public required int ExpectedPrice { get; set; }
    public required int Day { get; set; }
}

public sealed class ItineraryDto
{
    public required int Id { get; set; }
    public required int TripId { get; set; }
    public required string Title { get; set; }
    public string? Description { get; set; }
    public required int ExpectedPrice { get; set; }
    public required int Day { get; set; }
    public required int CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

public sealed class UpdateItineraryDto
{
    public required string Title { get; set; }
    public string? Description { get; set; }
    public required int ExpectedPrice { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public sealed class SearchItineraryDto
{
    public int? TripId { get; set; }
    public string? Title { get; set; }
    public string? Description { get; set; }
    public int? ExpectedPrice { get; set; }
    public int? Day { get; set; }
    public int? CreatedBy { get; set; }
}

File: frontend/src/app/core/feature/itinerary/dtos/itinerary.dto.ts
export interface ItineraryDto {
  id: number;
  tripId: number;
  title: string;
  description: string | null;
  expectedPrice: number;
  day: number;
  createdBy: number;
  createdAt: string;
  updatedAt: string | null;
}

export interface CreateItineraryDto {
  tripId: number;
  title: string;
  description?: string;
  expectedPrice: number;
  day: number;
}

export type ItinerarySearchFieldsDto = {
  tripId: number;
  day: number;
  title: string;
  expectedPrice: number;
  createdBy: number;
};

export interface TripTotalPriceDto {
	totalPrice: number;
}

export type ItineraryOrderByDto = 'id' | 'tripId' | 'title' | 'expectedPrice' | 'day' | 'createdBy';
